src/Datadog.Trace/HttpOverStreams/HttpContent/StreamContent.cs

```diff
@@ -21,9 +22,9 @@ public Task CopyToAsync(Stream destination, int maxBufferSize)
 
             if (Length != null)
             {
-                if (Length > maxBufferSize)
+                if (Length > int.MaxValue)
```

Comment:
Add this catch all simply because there is a variable type mismatch, I figured that the more descriptive error is worth it.